app/src/test/java/de/westnordost/streetcomplete/quests/surface/AddPitchSurfaceTest.kt

```diff
@@ -0,0 +1,90 @@
+package de.westnordost.streetcomplete.quests.surface
+
+import de.westnordost.osmapi.map.data.OsmNode
+import de.westnordost.osmapi.map.data.OsmWay
+import de.westnordost.streetcomplete.quests.TestMapDataWithGeometry
+import org.junit.Assert.*
+import org.junit.Test
+
+class AddPitchSurfaceTest {
+    private val questType = AddPitchSurface()
+
+    @Test fun `not applicable to pitch area without tagged sport`() {
+        val mapData = TestMapDataWithGeometry(listOf(
+            OsmWay(1L, 1, listOf(1,2,3), mapOf(
+                "leisure" to "pitch",
+            ))
+        ))
+        assertEquals(0, questType.getApplicableElements(mapData).toList().size)
+    }
```

Comment:
The test is unnecessarily complex. For `OsmFilterQuestType`s, `getApplicableElements` directly refers to `isApplicableTo`, also, you are testing very similar things, so lets add a few helper functions
```suggestion
    @Test fun `not applicable to pitch area without tagged sport`() {
        assertIsNotApplicable("leisure" to "pitch")
    }
    
    private fun assertIsApplicable(vararg vararg pairs: Pair<K, V>) {
        assertTrue(questType.isApplicableTo(create(*pairs)))
    }
    
    private fun assertIsNotApplicable(vararg vararg pairs: Pair<K, V>) {
        assertFalse(questType.isApplicableTo(create(*pairs)))
    }
    
    private fun create(vararg vararg pairs: Pair<K, V>) = OsmWay(1L, 1, listOf(1,2,3), mapOf(pairs))
```

If you do this for all the tests, you'll shrink it somewhat. Add more helper functions like

```kotlin

```

Reply:
Maybe even put it all in one test, I know it is considered not that good of a style but if the `assert...` statement already reads self-explanatory, it feels like unnecessary duplication to have the title of the test also in the function name.